lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -246,45 +246,43 @@ SmallVector<unsigned> getWarpOrder(Attribute layout) {
 
 SmallVector<unsigned> getOrder(Attribute layout) {
   if (auto blockedLayout = dyn_cast<BlockedEncodingAttr>(layout)) {
-    return SmallVector<unsigned>(blockedLayout.getOrder().begin(),
-                                 blockedLayout.getOrder().end());
-  } else if (auto mmaLayout = dyn_cast<MmaEncodingTrait>(layout)) {
+    return llvm::to_vector(blockedLayout.getOrder());
+  }
+  if (auto mmaLayout = dyn_cast<MmaEncodingTrait>(layout)) {
     auto distributedLayout = cast<DistributedEncodingTrait>(layout);
     auto rank = distributedLayout.getWarpsPerCTA().size();
     SmallVector<unsigned> order(rank);
-    for (auto i = 0; i < rank; ++i)
-      order[i] = rank - 1 - i;
+    std::iota(order.rbegin(), order.rend(), 0);
```

Comment:
I think you want to move this variable `order` into the `mfmaLayout` branch below?  The other branches declare their own `order` variable.

Reply:
not sure i follow exactly--this `order` is used for this whole case and we just have a fixup for it for mfma specifically.

Reply:
I may be reading the code wrong, but `if (dotLayout)` declares its own `order` variable, as does `if (sliceLayout)`.  So I don't think the `order` variable is used anywhere other than inside `if (fmaLayout)`?

Reply:
hmm, maybe the way GitHub renders diff is confusing? `order` is shared between mfma case vs the other `MmaEncodingTrait` cases; so cannot be privite to `mma` fixup branch. I reworked it a bit still though; hopefully it's better this time.

Reply:
okay, I may have been misreading the code then.  Anyway I trust you.  Thanks :)